### PR TITLE
Koboclara2 e

### PR DESCRIPTION
**Pull request by psa-avat**
Brief summary of the changes
----------------------------
Add the support of the new KOBO 2E and change the way to enumerate serial ports on Kobo.

Related issues and discussions
------------------------------

### CODE REVIEW

**Comment by lordfolken** [comment]
Can you rebase and remove the fixup commit vscode?

Commit mesage please: kobo: support for Kobo 2E

File: src/Kobo/Model.cpp [hunk 1]
```diff
@@ -94,5 +95,30 @@ DetectKoboModel() noexcept
...
+  KoboModel  model = DetectKoboModel() ;
+
+  if (model == KoboModel::LIBRA2 )
+    return "wlan0" ;
+  else 
+  if (model == KoboModel::CLARA_2E )
+    return "mlan0" ;
+    
+  return "eth0";
+}
```
Comment (lordfolken) [inline src/Kobo/Model.cpp:109]: please make this a switch statement with a default

File: src/Kobo/Model.hpp [hunk 2]
```diff
@@ -49,3 +50,7 @@ DetectKoboModel() noexcept;
 [[gnu::const]]
 const char *
 GetKoboWifiInterface() noexcept;
+
+[[gnu::const]]
+const char *
+GetKoboModel() noexcept;
```
Comment (lordfolken) [inline src/Kobo/Model.hpp:56]: missing new line

File: src/Kobo/Model.cpp [hunk 3]
```diff
@@ -94,5 +95,30 @@ DetectKoboModel() noexcept
...
+GetKoboModel() noexcept
+{
+  static char buffer[16];
+  if (!ReadFromFile("/dev/mmcblk0", 0x200, buffer, sizeof(buffer)))
+    buffer[0]=0 ;
+  
+  // trunc the string to 8 char
+  buffer[8]=0 ;
+
+  return buffer ;
```
Comment (lordfolken) [inline src/Kobo/Model.cpp:122]: code style, why spaces between statement and semi-colon?

File: src/system/FileUtil.cpp [hunk 4]
```diff
@@ -314,8 +319,24 @@ File::Exists(Path path) noexcept
...
+  int fd = open (path.c_str(), O_RDWR | O_NONBLOCK) ;
+  if (fd < 0) return false ;
+
+  struct serial_struct serinfo ;
+  int ret = ioctl (fd, TIOCGSERIAL, &serinfo) ;
+  close(fd) ;
+  if (ret >=0) return true ;
+  return false;
+  
+#else
```
Comment (lordfolken) [inline src/system/FileUtil.cpp:335]: formatting, spaces

File: src/Kobo/System.cpp [hunk 5]
```diff
@@ -255,12 +263,31 @@ KoboWifiOn()
...
+
+    case KoboModel::LIBRA2 :
+    case KoboModel::CLARA_2E :
+      driver = "nl80211" ;
+      break ;
+    
+    default : 
+      break ;
+
+  }
```
Comment (lordfolken) [inline src/Kobo/System.cpp:289]: spaces, unnecessary new-lines

File: src/Kobo/SystemDialog.cpp [hunk 6]
```diff
@@ -74,8 +76,16 @@ void
...
+
+  // Suppres OTG for CLARA HD 2E. It crashes the KOBO and implies a flash of SD Card
+  if ( DetectKoboModel() != KoboModel::CLARA_2E)
+  {
+    switch_otg_mode = AddButton(IsKoboOTGHostMode() ? "Disable USB-OTG" : "Enable USB-OTG",
             [this](){ SwitchOTGMode(); });
+  }
+  else 
+    AddDummy();
+
```
Comment (lordfolken) [inline src/Kobo/SystemDialog.cpp:88]: Doesn't that break OTG support for all other models?

**Comment by lordfolken** [comment]
@GeekGoneOld can you also please do a review?

**Comment by ubx** [comment]
This PR is obsolete. Support for Kobo Clara 2E was added here cee33f224f076ad2a52cce5ea9c246eef017ebc4

**Comment by MaxKellermann** [comment]
Sigh. The third PR by @psa-avat in a row which made no sense at all, which was 100% broken.

**Pull request closed without merging**